content/dev/plugins/help.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Building help for your admin plugin 
+description: Creating easy-to-access help for your Zen Cart admin-side modification
+category: plugins
+weight: 10
+---
+
+## Help for New Admin Pages 
+
+If your plugin creates a new page, the ability to provide help has been built-in since Zen Cart 1.5.8.  If you are not yet using this version or higher, you can merge [this PR](https://github.com/zencart/zencart/pull/4243/commits/ea92e4a950ba0c373ec081a3472d16f5030a70c1) into your cart. 
+
+Use the [observer autoloading](https://docs.zen-cart.com/dev/code/notifiers/#auto-loaded-observers) feature of Zen Cart admin observers (available since Zen Cart 1.5.7), and create a new file called `admin/includes/classes/observers/auto.MyPluginHelp.php`.  (Be sure to follow the naming conventions for observer autoloading!)
+
+```
+<?php
+class zcObserverMyPluginHelp extends base {
+  function __construct() {
+    $this->attach($this, array('NOTIFIER_PLUGIN_HELP_PAGE_URL_LOOKUP'));
+  }
+
+  function update(&$class, $eventID, $page, &$help_page) {
+     if ($eventID == 'NOTIFIER_PLUGIN_HELP_PAGE_URL_LOOKUP') {
+        if ($page == FILENAME_MYPLUGIN_NAME) {
+           $help_page = "help/myplugin.html"; 
+        }
+     }
+  }
+}
+```
+
+Then create a help file in `YOURADMIN/help/myplugin.html` (or `YOURADMIN/help/myplugin.php` if you need to execute code) to store your help text.
```

Comment:
CAUTION: That's an entirely new directory, which needs to be protected from misuse and from unauthorized content storage.
Here be dragons.

Reply:
Put at top level instead?  I am open to other approaches.

Reply:
I'm not saying don't use such a directory. I'm saying that if you do then be sure to teach how to secure it. At a basic level perhaps copy the default `index.html` and `index.php` files from the `/cache` dir into this `/help` dir.

Or build the plugin intuitively enough that none of this is needed. :D

One must expect that bad-actors will do everything they can to enumerate the contents of every directory they can find, hoping to discover any clues about potential vulnerabilities they could try to exploit. 
Every directory that's publicly accessible must be treated with caution. Always assume all of its contents will be exposed to your worst enemy and they will use it against you. Therefore never put anything sensitive there. In fact, put as little as possible there. Less is ALWAYS more.

The danger here is even doing it at all.

This is why we always encourage people to NEVER upload `/docs` files and README files and plugin documentation directories to their server.

Reply:
Removing this content on next checkin.